pkg/cmd/codespace/common.go

```diff
@@ -31,23 +31,35 @@ type executable interface {
 	Executable() string
 }
 
+type progressIndicator interface {
+	StartProgressIndicatorWithLabel(s string)
+	StopProgressIndicator()
+}
+
+type statusChecker func(progressIndicator, codespaces.LiveshareApiClient, context.Context, *api.Codespace) error
+type sshClient func(app *App, ctx context.Context, sshArgs []string, opts sshOptions) (err error)
+
 type App struct {
-	io         *iostreams.IOStreams
-	apiClient  apiClient
-	errLogger  *log.Logger
-	executable executable
-	browser    browser
+	io            *iostreams.IOStreams
+	apiClient     apiClient
+	errLogger     *log.Logger
+	executable    executable
+	browser       browser
+	sshClient     sshClient
+	statusChecker statusChecker
```

Comment:
I understand this was added to avoid exercising `App.showStatus()`, which currently doesn't seem to have test coverage. Could we instead somehow mock `codespaces.PollPostCreateStates()` (a sub-dependency of `showStatus()` and let the real `showStatus` it be exercised?